weave/type_handlers/Video/video.py

```diff
@@ -0,0 +1,135 @@
+"""Defines the custom Video weave type."""
+
+from __future__ import annotations
+
+import os
+import shutil
+from typing import TYPE_CHECKING, Any
+
+from weave.trace.serialization import serializer
+from weave.trace.serialization.custom_objs import MemTraceFilesArtifact
+
+try:
+    from moviepy.editor import (
+        VideoClip,
+        VideoFileClip,
+    )
+except ImportError:
+    dependencies_met = False
+else:
+    dependencies_met = True
+
+if TYPE_CHECKING:
+    from moviepy.editor import (
+        VideoClip,
+        VideoFileClip,
+    )
+
+SUPPORTED_FORMATS = ["gif", "mp4", "webm"]
```

Comment:
Maybe use an enum here?

Reply:
I made an enum and constructed the supported formats from its values. Should make it easier/clearer when adding or removing formats